lib/Dialect/FIRRTL/Transforms/EmitMetadata.cpp

```diff
@@ -0,0 +1,275 @@
+//===- EmitMetadata.cpp - Emit various types of metadata --------*- C++ -*-===//
```

Comment:
Similar, if this is sifive specific, please name "EmitSiFiveMetadata".  Alternatively, would "Generate" or "Create"SiFiveMetadata be better - emitters sound like they print something, but this is creating IR nodes that are eventually printed.